oxo.py

```diff
@@ -1,36 +1,80 @@
 class OxoBoard:
+
     def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.grid_width = 3  # Stores number of columns in class
+        self.grid_height = 3  # Stores number of rows in class
+        self.win_length = 3  # Stores win_length in class
+        self.board = [[0 for x in xrange(0, self.grid_width)] for y in xrange(0, self.grid_height)]
+        # Sets a list to hold the board state
+        self.show()
+        self.last_x = 0
+        self.last_y = 0
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x][y]  # Returns board state at x,y
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.board[x][y] == 0:
+            self.last_x = x
+            self.last_y = y
+            self.board[x][y] = mark  # Sets current square to the given mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        board_state = True  # Changed to False if the board in not full
+        for x in xrange(0, self.grid_width):  # Loops check every square on the board
+            for y in xrange(0, self.grid_height):
+                if self.board[x][y] == 0:
+                    board_state = False  # If a square is empty change board_state
```

Comment:
You could just `return False` here, as there is no way for `board_state` to become `True` again. Would be a little more efficient